### PR TITLE
More helpful semver range message

### PR DESCRIPTION
**Pull request by bsyouness**
As per #3178

### CODE REVIEW

File: jupyterlab_widgets/pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab~=3.0", "setuptools>=40.8.0", "wheel"]
```
Comment (ianhi) [inline jupyterlab_widgets/pyproject.toml:2]: Why these changes? I believe that the pin on `jupyter_packaging` is important as the API for that package is subject to change.

Comment (bsyouness) [reply jupyterlab_widgets/pyproject.toml:2]: The build was broken before these changes (see commit 2b9a2d74). I didn't want to pin a version of jupyter_packaging but 7.9 doesn't have get_version_info, so I just left it out. I can add a TODO to figure out what version of jupyter_packaging to use given that it's out of scope for this PR. (I think get_version_info was introduced with 9.0.)

Comment (jtpio) [reply jupyterlab_widgets/pyproject.toml:2]: Maybe we could then update to jupyter-packaging 0.9 in a separate PR so it's easier to track?

Comment (jtpio) [reply jupyterlab_widgets/pyproject.toml:2]: Opened https://github.com/jupyter-widgets/ipywidgets/pull/3194

Comment (bsyouness) [reply jupyterlab_widgets/pyproject.toml:2]: Thanks!

**Comment by bsyouness** [comment]
@jasongrout now that #3194 is merged, this should be ready to be merged as well.

**Comment by jasongrout** [comment]
Thanks. Adding it to my queue (but @jtpio and @ianhi, feel free to merge if you've already reviewed this)

File: jupyterlab_widgets/src/semvercache.ts [hunk 2]
```diff
@@ -28,6 +28,12 @@ export class SemVerCache<T> {
     }
   }
 
+  getAllModules(key: string): Record<string, any> | undefined {
```
Comment (jasongrout) [inline jupyterlab_widgets/src/semvercache.ts:31]: Can we change this to `getAllVersions`? I think that would be a bit clearer about what is being returned.

**Review by jasongrout — APPROVED** [review]
This looks good, with one change to the API to be a bit clearer (in my opinion - if you don't like the change, please discuss).

**Comment by bsyouness** [comment]
There is something broken with the auto-linting. I only pushed my changes with the new name for `getAllVersions` and it triggered an auto-linting event and the linter check complains about the changes that were made by the auto-linter...

**Comment by jtpio** [comment]
> There is something broken with the auto-linting. I only pushed my changes with the new name for `getAllVersions` and it triggered an auto-linting event and the linter check complains about the changes that were made by the auto-linter...

`prettier` was updated in https://github.com/jupyter-widgets/ipywidgets/pull/3162. So a git clean and reinstalling the dependencies from scratch should hopefully help.

**Comment by bsyouness** [comment]
Should be good to merge now.

**Review by jtpio — APPROVED** [review]
Thanks!
